### PR TITLE
fetch running workflow

### PR DESCRIPTION
**Pull request by Yancey0623**
part work of #1126

### CODE REVIEW

**Review by wangkuiyi — CHANGES_REQUESTED** [review]
Please file the design doc PR before any code change.

**Comment by Yancey0623** [comment]
> Please file the design doc PR before any code change.

Thank you for reminding me, I will file a design doc today, and then finish this PR.

File: pkg/argo/fetch.go [hunk 1]
```diff
@@ -45,38 +43,57 @@ func newFetchResponse(newReq *pb.FetchRequest, eof bool, logs []string) *pb.Fetc
...
 func Fetch(req *pb.FetchRequest) (*pb.FetchResponse, error) {
-	// TODO(yancey1989): fetching running workflow logs
-	wf, err := waitUntilComplete(req)
+	wf, err := k8sReadWorkflow(req.Job.Id)
 	if err != nil {
 		return nil, err
 	}
 
-	stepGroupName, err := getCurrentStepGroup(wf, req.Job.Id, req.StepId)
+	if isWorkflowPending(wf) {
```
Comment (typhoonzero) [inline pkg/argo/fetch.go:51]: Is there a way to add test to cover wokflow/pod pending branches?

**Review by typhoonzero — COMMENTED** [review]
Basically LGTM

**Review by wangkuiyi — APPROVED** [review]

**Comment by wangkuiyi** [comment]
I am approving following @typhoonzero 's comments.

**Review by tonyyang-svail — APPROVED** [review]
LGTM.

File: pkg/argo/fetch.go (hunk 1 above)
Comment (Yancey0623) [reply pkg/argo/fetch.go:51]: Good point, will do that in the next PR.

**Pull request merged**